src/mantine-form/src/resolvers/zod-resolver/zod-resolver.ts

```diff
@@ -5,30 +5,32 @@ interface ZodError {
   message: string;
 }
 
-interface ZodResults {
-  success: boolean;
+interface ZodParseSuccess {
+  success: true;
+}
+
+interface ZodParseError {
+  success: false;
   error: {
     errors: ZodError[];
   };
 }
 
-interface ZodSchema {
-  safeParse(values: Record<string, any>): ZodResults;
+interface ZodSchema<T extends Record<string, any>> {
```

Comment:
Just wondering, is using the exported zod types not a viable option here?

Reply:
No, we don't want Mantine users to have Zod as dependency/devDep just for the type.